File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int i = 0, j = numbers.length - 1;
        while (i < numbers.length && j >=0)    {
            int sum  = numbers[i] + numbers[j];
            if (sum < target)   {
                i++;
            } else if (sum > target) {
                j--;
            } else  {
                return new int[]{i + 1, j + 1};
            }
        }
        return new int[]{};
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        soln_map = {}

        for i in range(len(nums)):
            complement = target - nums[i]

            if complement in soln_map:
                return [soln_map[complement], i]
            else:
                soln_map[nums[i]] = i